File: springboot-kotlin/src/main/kotlin/cn/danao/springboot/springbootkotlin/controller/StartController.java
package cn.danao.springboot.springbootkotlin.controller;

import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * date 2020/9/2 17:39 <br/>
 * description class <br/>
 *
 * @author danao
 * @version 1.0
 * @since 1.0
 */
@RestController
@RequestMapping(value = "/start")
@Slf4j
public class StartController {

    /**
     * http://127.0.0.1:8080/start/test?key=key
     *
     * @param key
     * @return
     */
    @GetMapping(value = "/test")
    public String start(String key) {
        log.info("传入参数={}", key);
        String result = "{\n" +
                "  \"status\": true\n" +
                "}";
        return result;
    }

}

File: kotlin_project/src/main/kotlin/cn/danao/kotlin_project/controller/StartController.kt
package cn.danao.kotlin_project.controller

import cn.danao.kotlin_project.server.StartServer
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
//kotlin目前不能使用@Slfj注解
//@Slf4j
@RequestMapping(value = ["/start"])
class StartController {

    @Autowired
    lateinit var startServer: StartServer

    fun isInstall() = ::startServer.isInitialized

    //创建日志
    var log = LoggerFactory.getLogger(this.javaClass)

    @RequestMapping(value = ["/test"])
    fun testRequest(name: String): String {
        log.info("name = ${name}")
        println("name = ${name}")
        var sum = startServer.getSum(12, 14)
        var test = startServer.interfaceTest()
        log.info("sum = ${sum}, test = ${test}")
        return "测试"
    }

}